--- snappy-1.0.1/Makefile.am.orig	2011-03-25 00:20:16.000000000 +0100
+++ snappy-1.0.1/Makefile.am	2011-04-04 18:41:40.079842328 +0200
@@ -12,7 +12,7 @@ noinst_HEADERS = snappy-internal.h snapp
 snappy_unittest_CPPFLAGS = $(gflags_CFLAGS) $(GTEST_CPPFLAGS)
 snappy_unittest_SOURCES = snappy_unittest.cc snappy-test.cc
 snappy_unittest_LDFLAGS = $(GTEST_LDFLAGS)
-snappy_unittest_LDADD = libsnappy.la $(gflags_LIBS) $(GTEST_LIBS)
+snappy_unittest_LDADD = libsnappy.la $(gflags_LIBS) $(GTEST_LIBS) $(OTHERS_LIBS)
 TESTS = snappy_unittest
 noinst_PROGRAMS = $(TESTS)
 
--- snappy-1.0.1/configure.ac.orig	2011-03-25 00:20:16.000000000 +0100
+++ snappy-1.0.1/configure.ac	2011-04-04 18:42:55.415849032 +0200
@@ -79,11 +79,13 @@ fi
 
 # Other compression libraries; the unit test can use these for comparison
 # if they are available. If they are not found, just ignore.
-AC_CHECK_LIB([z], [zlibVersion],, [true])
-AC_CHECK_LIB([lzo2], [lzo1x_1_15_compress],, [true])
-AC_CHECK_LIB([lzf], [lzf_compress],, [true])
-AC_CHECK_LIB([fastlz], [fastlz_compress],, [true])
-AC_CHECK_LIB([quicklz], [qlz_compress],, [true])
+OTHERS_LIBS=
+AC_CHECK_LIB([z], [zlibVersion], [OTHERS_LIBS="$OTHERS_LIBS -lz"], [true])
+AC_CHECK_LIB([lzo2], [lzo1x_1_15_compress], [OTHERS_LIBS="$OTHERS_LIBS -llzo2"], [true])
+AC_CHECK_LIB([lzf], [lzf_compress], [OTHERS_LIBS="$OTHERS_LIBS -llzf"], [true])
+AC_CHECK_LIB([fastlz], [fastlz_compress], [OTHERS_LIBS="$OTHERS_LIBS -lfastlz"], [true])
+AC_CHECK_LIB([quicklz], [qlz_compress], [OTHERS_LIBS="$OTHERS_LIBS -lquicklz"], [true])
+AC_SUBST([OTHERS_LIBS])
 
 # These are used by snappy-stubs-public.h.in.
 if test "$ac_cv_header_stdint_h" = "yes"; then
