--- snappy-1.1.9/snappy.cc.orig	2021-05-05 00:53:34.000000000 +0200
+++ snappy-1.1.9/snappy.cc	2022-02-14 16:46:13.219517746 +0100
@@ -1014,7 +1014,7 @@ void MemMove(ptrdiff_t dst, const void*
 }
 
 SNAPPY_ATTRIBUTE_ALWAYS_INLINE
-size_t AdvanceToNextTag(const uint8_t** ip_p, size_t* tag) {
+static inline size_t AdvanceToNextTag(const uint8_t** ip_p, size_t* tag) {
   const uint8_t*& ip = *ip_p;
   // This section is crucial for the throughput of the decompression loop.
   // The latency of an iteration is fundamentally constrained by the
